### DISCUSSION TITLE
ARW too bright converted to PNG in Imagemagick

### DISCUSSION THREAD

**Discussion by fmw42** [ImageMagick/ImageMagick#6748] [Digital Image Processing]
I have this ARW image (from question on StackOverflow). The complaint was that when converting with other tool (ConvertIO online) it looks about the same as the ARW. But converted with Imagemagick, it is too bright. I have not been able to find or use properly any define to make it look more normal. It looks like it has been brightness enhanced.

Here is the original:

[DSC04420.ARW.zip](https://github.com/ImageMagick/ImageMagick/files/12772816/DSC04420.ARW.zip)

Here is the result from https://convertio.co converted from PNG to JPG as the PNG was too big to post.

![DSC04420_convertio](https://github.com/ImageMagick/ImageMagick/assets/13458218/41cb1c39-c4c6-4c9a-acd2-24f07749ddac)

Here are the commands that I tried:

`magick DNG:DSC04420.ARW -define dng:output-color=0 -define dng:use-camera-wb=false -define dng:use-auto-wb=false -define dng:use-auto-bright=false DSC04420_imagemagick.png`

Also tried dng:output-color=1,2,3,4,5 and 6 with little change. And also tried putting the defines before reading the input.

Here is the result from that command (converted from PNG to JPG):

![DSC04420_imagemagick](https://github.com/ImageMagick/ImageMagick/assets/13458218/9b62eaf0-ffab-4be7-a7a9-6a25ffe30e4e)

I rarely use raw formats, so request help from anyone who does.

**Comment by snibgo** [comment]
All those `-define` options need to be *before* reading the raw file. IM performs operations in the order you give them. The `-define` options would affect the read of the input raw, provided they are given first.

You have:
```
-define dng:use-auto-bright=false
```
... but that option isn't listed on https://imagemagick.org/script/defines.php . Defining options that are meaningless doesn't raise an error, because IM doesn't know they are meaningless.

Instead, you could have:
```
-define dng:no-auto-bright=true
```

**Reply by fmw42** [reply]
That option is listed on the formats page.  I guess it was overlooked in the defines page.  See https://imagemagick.org/script/formats.php

I had tried putting the defines after and repeated with them before reading the image, but no difference.

I tried all these commands again and none made a difference:

```
magick -define dng:use-camera-wb=true DNG:DSC04420.ARW DSC04420.png

magick -define dng:use-camera-wb=false DNG:DSC04420.ARW DSC04420.png

magick -define dng:use-camera-wb=true -define dng:use-auto-wb=false -define dng:use-auto-bright=false DNG:DSC04420.ARW DSC04420.png

magick -define dng:use-camera-wb=false -define dng:use-auto-wb=false -define dng:use-auto-bright=false DNG:DSC04420.ARW DSC04420.png

```

**Comment by snibgo** [comment]
Thanks, fmw42. Yes, the formats page https://imagemagick.org/script/formats.php mentions `use-auto-bright`. That is a documentation bug. IM source code does not contain that string. Please change that documentation to `no-auto-bright`.

Auto brightness is on by default. That is the problem shown in the OP, where auto brightness is not wanted. To turn auto brightness off, use this before reading the input raw file:
```
-define dng:no-auto-bright=true
```

**Comment by urban-warrior** [comment]
@snibgo is correct, its `no-auto-bright`.  We corrected the documentation.

**Reply by fmw42** [reply]
Thanks, that works in Imagemagick 7.  But it does not work in Imagemagick 6.  Might this be changed to work there also.

And perhaps put a reminder on the DNG defines that they need to be put before reading the input.

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce the issue and will have a patch to fix it in the GIT main branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://imagemagick.org/archive/beta/ by sometime tomorrow.

**Reply by fmw42** [reply]
Thanks.  This seems to be fixed in the 6.9.12-97
